mldsa/sign.c

```diff
@@ -13,18 +14,20 @@
 #include "symmetric.h"
 
 /*************************************************
- * Name:        crypto_sign_keypair
+ * Name:        crypto_sign_keypair_internal
  *
- * Description: Generates public and private key.
+ * Description: FIPS 204: Algorithm 6 ML-DSA.KeyGen_internal.
+ *              Generates public and private key. Internal API.
  *
- * Arguments:   - uint8_t *pk: pointer to output public key (allocated
- *                             array of CRYPTO_PUBLICKEYBYTES bytes)
- *              - uint8_t *sk: pointer to output private key (allocated
- *                             array of CRYPTO_SECRETKEYBYTES bytes)
+ * Arguments:   - uint8_t *pk:   pointer to output public key (allocated
+ *                               array of CRYPTO_PUBLICKEYBYTES bytes)
+ *              - uint8_t *sk:   pointer to output private key (allocated
+ *                               array of CRYPTO_SECRETKEYBYTES bytes)
+ *              - uint8_t *seed: pointer to input random seed (SEEDBYTES bytes)
  *
  * Returns 0 (success)
  **************************************************/
-int crypto_sign_keypair(uint8_t *pk, uint8_t *sk)
+int crypto_sign_keypair_internal(uint8_t *pk, uint8_t *sk, uint8_t *seed)
```

Comment:
Didn't spot early enough, this should have been `const uint8_t *seed`. @mkannwischer you did this right in #38, so let's fix it in the rebase of #38.